apps/wallet/src/ui/app/wallet/hooks.tsx

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useEffect } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+import { WALLET_ENCRYPTION_ENABLED } from './constants';
+import { useAppSelector } from '_hooks';
+
+export function useLockedGuard(requiredLockedStatus: boolean) {
```

Comment:
I think a better approach would be to do this at the router layout level. It should be easy to put a bunch of pages under a layout that checks the locked state and conditionally renders either an `<Ouetlet />` or a redirect based on that state. That would let pages be completely un-aware of locking at all.

Reply:
I think it's very close to what you describe already. Only 3 places are using it `HomePage` `Initialise` and  `Locked` that are kind of a wrapper page (except the Locked page). I think the way it's done gives some flexibility but makes sense what you say. I will create a ticket for this and maybe discuss to do something similar with the other guards.